src/mscorlib/src/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -796,25 +800,29 @@ internal unsafe int GetHashCodeOfStringCore(string source, CompareOptions option
 
             int sortKeyLength = Interop.Globalization.GetSortKey(_sortHandle, source, source.Length, null, 0, options);
 
```

Comment:
Presumably we could supply a buffer that would often be big enough and avoid the need to call Getsortkey twice.

Reply:
Yes, there are many performance improvements that can be done in the globalization implementation. Not doing them in this PR to make it scoped.

Reply:
Hashing is possibly the most perf sensitive path in globalization? Perhaps it's worth an issue opened.